src/BaseMailer.php

```diff
@@ -0,0 +1,247 @@
+     * The callback should return a file name which will be used to save the email message.
+     * If not set, the file name will be generated based on the current timestamp.
+     *
+     * The signature of the callback is:
+     *
+     * ```php
+     * function ($mailer, $message)
+     * ```
+     */
+    public $fileTransportCallback;
+
+    /**
+     * @var MessageFactoryInterface $messageFactory
+     */
+    private $messageFactory;
+
+    /**
+     * @var Composer $composer Message composer instance.
+     */
+    private $composer;
+
+    /**
+     * @var EventDispatcherInterface $eventDispatcher.
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var LoggerInterface $logger.
+     */
+    private $logger;
+
+    /**
+     * @param MessageFactoryInterface $messageFactory
+     * @param Composer $composer
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param LoggerInterface $logger
+     */
+    public function __construct(MessageFactoryInterface $messageFactory, Composer $composer, EventDispatcherInterface $eventDispatcher, LoggerInterface $logger)
+    {
+        $this->messageFactory = $messageFactory;
+        $this->composer = $composer;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return Composer message composer instance.
+     */
+    public function getComposer(): Composer
+    {
+        return $this->composer;
+    }
+
+    /**
+     * Sets message composer.
+     * 
+     * @param Composer $composer message composer instance.
+     */
+    public function setComposer(Composer $composer)
+    {
+        $this->composer = $composer;
+    }
+
+    /**
+     * Creates a new message instance and optionally composes its body content via view rendering.
+     *
+     * @param string|array|null $view the view to be used for rendering the message body. This can be:
+     *
+     * - a string, which represents the view name for rendering the HTML body of the email.
+     *   In this case, the text body will be generated by applying `strip_tags()` to the HTML body.
+     * - an array with 'html' and/or 'text' elements. The 'html' element refers to the view name
+     *   for rendering the HTML body, while 'text' element is for rendering the text body. For example,
+     *   `['html' => 'contact-html', 'text' => 'contact-text']`.
+     * - null, meaning the message instance will be returned without body content.
+     *
+     * The view to be rendered can be specified in one of the following formats:
+     *
+     * - a relative view name (e.g. "contact") located under [[viewPath]].
+     *
+     * @param array $params the parameters (name-value pairs) that will be extracted and made available in the view file.
+     * @return MessageInterface message instance.
+     */
+    public function compose($view = null, array $params = []): MessageInterface
```

Comment:
```suggestion
    public function compose(View $view = null, array $params = []): MessageInterface
```

Reply:
The parameter `$view` here is not an instance of `View`.